eng/build.ps1

```diff
@@ -109,3 +110,36 @@ if ($env:TreatWarningsAsErrors -eq 'false') {
 
 Write-Host "& `"$PSScriptRoot/common/build.ps1`" $arguments"
 Invoke-Expression "& `"$PSScriptRoot/common/build.ps1`" $arguments"
+$buildExitCode = $LASTEXITCODE
+
+# Install MAUI workload after restore if -restore-maui was passed
+# Only on Windows and macOS (MAUI doesn't support Linux)
+$restoreMauiPassed = $properties -contains "-restore-maui"
```

Comment:
@jfversluis is the plan to also add a github workflow or something that is protecting this part of the script and ensuring it doesn't get broken? We should have some sort of protection by having a job that restores the workload and tries to build the maui playground project.

Reply:
Absolutely! Opened this to track it for now: https://github.com/dotnet/aspire/issues/12372